bridge/client/app/_components/ktb-create-secret-form/ktb-create-secret-form.component.ts

```diff
@@ -1,57 +1,67 @@
-import {Component, OnInit} from '@angular/core';
-import {DataService} from '../../_services/data.service';
-import {ActivatedRoute, Router} from '@angular/router';
-import { AbstractControl, FormControl, FormGroup, Validators } from '@angular/forms';
-import {Secret} from '../../_models/secret';
+import { Component } from '@angular/core';
+import { DataService } from '../../_services/data.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AbstractControl, FormArray, FormBuilder, Validators } from '@angular/forms';
+import { Secret } from '../../_models/secret';
 
 @Component({
   selector: 'ktb-secrets-view',
   templateUrl: './ktb-create-secret-form.component.html',
   styleUrls: ['./ktb-create-secret-form.component.scss']
 })
-export class KtbCreateSecretFormComponent implements OnInit {
+export class KtbCreateSecretFormComponent {
+
   public isLoading = false;
-  public secret: Secret = new Secret();
+
   private secretNamePattern = '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*';
   private secretKeyPattern = '[-._a-zA-Z0-9]+';
 
-  public defaultFormControls: {} = {
-    name: new FormControl('', [Validators.required, Validators.pattern(this.secretNamePattern)])
-  };
-  public createSecretForm = new FormGroup(this.defaultFormControls);
+  public createSecretForm = this.fb.group({
+    name: ['', [Validators.required, Validators.pattern(this.secretNamePattern)]],
+    data: this.fb.array([
+      this.fb.group({
+        key: ['', [Validators.required, Validators.pattern(this.secretKeyPattern)]],
+        value: ['', [Validators.required]]
+      })
+    ])
+  });
 
-  constructor(private dataService: DataService, private router: Router, private route: ActivatedRoute) { }
+  constructor(private dataService: DataService, private router: Router, private route: ActivatedRoute, private fb: FormBuilder) {
+  }
 
-  ngOnInit(): void {
-    this.secret = new Secret();
-    this.addPair();
+  get data(): FormArray | null {
+    return this.createSecretForm.get('data') as FormArray;
```

Comment:
Please remove the code smell here :) I think `as FormArray` is not necessary.

Reply:
Because of the initialization of `createSecretForm` we can say it is not null. So the solution here would be to remove `| null` from the return type because it is not needed if we cast it to FormArray anyways.

Reply:
Without `as FormArray` TypeScript complains that `Type 'AbstractControl | null' is not assignable to type 'FormArray | null'.`
With it Sonar complains that `assertion is unnecessary since it does not change the type of the expression`.

Any ideas? Maybe @Kirdock has a suggestion?